File: 1_Two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int l=nums.length;
        HashMap<Integer,Integer> map=new HashMap<Integer,Integer>();
        for(int x=0;x<l;x++)
        {
            if(map.containsKey(target-nums[x]))
            {
                /*arr[0]=map.get(target-nums[x]);
                arr[1]=x;
                break;*/
                return new int[]{map.get(target-nums[x]),x};
            }
            else
            {
                map.put(nums[x],x);
            }
        }
        return new int[]{};
    }
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for idx,el in enumerate(nums):
            diff = target - el
            if(diff in map):
                return [map[diff],idx]
            else:
                map[el] = idx